### DISCUSSION TITLE
Grrlib examples for the Gamecube

### DISCUSSION THREAD

**Discussion by shazz** [GRRLIB/GRRLIB#26] [General]
Hi there!
I modified the examples to work on the gamecube but for some that's just a black screen and I cannot find why.
The culprits:
- `bitmap_fx`: black screen
- `funsin`: black screen
- `unlimited3d` (red flashing screen, buggy cubes) => memory issue ?

the ports are available [here!](https://github.com/PacMooseIncorporated/GameCube4Fun/tree/main/grrlib/examples_gc)

**Comment by Crayon2000** [comment]
Hello @shazz, I tried to generate those projects in the [gamecube-examples](https://github.com/GRRLIB/GRRLIB/tree/gamecube-examples) branch and tested it in Dolphin. The 3 examples you are mentioning seems to works in the emulator. I don't own a GameCube, would it be possible to test what was generated by the GitHub Actions: https://github.com/GRRLIB/GRRLIB/suites/6533673204/artifacts/243639879

Look into the _/gamecube_ folder. Try the .dol files (or .elf files if not working). Thanks 😀

**Comment by shazz** [comment]
Salut @Crayon2000 :)
Interesting... your funsin works! I'll check why as we apply the same modifications (rules and pad mgt)

Antoine's unlimited3 doesn't work, like for me, I guess it is a similar problem that in his unlimited2d screen. It was a little more complicated to patch, from my understanding, that was a memory issue due to the fact this cheater (kidding) was initializing 10 textures and it looks like on the GC it doesn't work with more than 3

````c
    GRRLIB_texImg *tex_screen[10];
    for(i=0; i<10; i++) {
        tex_screen[i] = GRRLIB_CreateEmptyTexture(rmode->fbWidth, rmode->efbHeight);
    }
````

Anyway, thanks for having spent the time to patch/build them! I'll give a second look!

**Comment by shazz** [comment]
This is pretty weird.... by recompiling your version I have the same problem, black screen. Does the [gamecube-examples](https://github.com/GRRLIB/GRRLIB/tree/gamecube-examples) branch contains GRRLIB modifications for the gamecube not in the master branch ? 
Or specific version of libogc ? 
Or devitpro_ppc ? 
Or some black magic ? :D

**Comment by Crayon2000** [comment]
The GitHub Action (_ci.yml_) is using the **devkitpro/devkitppc:latest** Docker image. Currently it comes with **devkitPPC r40** and **libogc 2.3.1**.
My [gamecube-examples](https://github.com/GRRLIB/GRRLIB/tree/gamecube-examples) branch is based on the latest master branch commit.
After that, I simply do:
```shell
        (cd GRRLIB && make clean all install)
        (cd GRRLIB/GRRLIB && make PLATFORM=cube clean all install)
        (cd examples && make)
```
So... no magic 🪄

**Reply by shazz** [reply]
Yeah I saw that... and I'm using the same docker image... but maybe I updated libogc, I'll check.

Could you also build the various grrlib demos? Or they only work on a wii?

**Comment by Crayon2000** [comment]
I have added to the [gamecube-examples](https://github.com/GRRLIB/GRRLIB/tree/gamecube-examples) branch all examples that don't requires IR sensor. Maybe at one point I will modify the code to make them work with the analog stick.

A new zip file is available at https://github.com/GRRLIB/GRRLIB/suites/6552258305/artifacts/244802325

If they mostly work, I will merge the branch back to the master branch so you can fix the GameCube examples with a PR (if you want 😁).

**Reply by shazz** [reply]
So, I tested them all this weekend, except `unlimited2d`, `unlimited3d` they all work!

**Comment by Crayon2000** [comment]
@shazz , I just fixed the makefile for `unlimited3d`. I also decrease the count of `tex_screen_count` in the `unlimited2d`. Can you try it again please: https://github.com/GRRLIB/GRRLIB/suites/6655671694/artifacts/251617137

**Reply by shazz** [reply]
I'll do that asap for sure!
Did you see my post on a bug I found and fixed?

**Reply by shazz** [reply]
Just tried, black screen for the 2d version and buggy for the 3d version : ![image](https://user-images.githubusercontent.com/604708/170697778-1b2fefea-7203-4cd0-8669-274cf305e223.jpeg)

**Reply by shazz** [reply]
my working 2d version uses:
```c 
const int tex_screen_count = 3;
```

instead of 4

**Comment by Crayon2000** [comment]
I have changed the unlimited 2D example with `tex_screen_count` set to 3. I have also changed the 2 unlimited examples Makefile.
The _gamecube-examples_ branch was merged back to master branch in PR #28.
Latest examples download is available here: https://github.com/GRRLIB/GRRLIB/suites/6698186012/artifacts/254465835
